component/common/net/config.go

```diff
@@ -3,30 +3,62 @@
 package net
 
 import (
-	"flag"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/middleware"
 	"time"
-
-	weaveworks "github.com/weaveworks/common/server"
 )
 
 const (
 	DefaultHTTPPort = 8080
-	DefaultGRPCPort = 8081
 )
 
-// ServerConfig is a River configuration that allows one to configure a weaveworks.Server. It
-// exposes a subset of the available configurations.
+// ServerConfig for a Server
 type ServerConfig struct {
+
+	// ==== configuration exposed in River configs ===
+
 	// HTTP configures the HTTP weaveworks. Note that despite the block being present or not,
 	// the weaveworks is always started.
 	HTTP *HTTPConfig `river:"http,block,optional"`
 
-	// GRPC configures the gRPC weaveworks. Note that despite the block being present or not,
-	// the weaveworks is always started.
-	GRPC *GRPCConfig `river:"grpc,block,optional"`
-
 	// GracefulShutdownTimeout configures a timeout to gracefully shut down the server.
 	GracefulShutdownTimeout time.Duration `river:"graceful_shutdown_timeout,attr,optional"`
+
+	// ==== configuration NOT exposed in River configs ===
+
```

Comment:
Exposing these in River would be trivial now and all components could benefit from new configuration options, e.g. TLS.